### PR TITLE
Fix crash when setBreakpoint from VSCode sends a git:/ URI...

### PR DESCRIPTION
**Pull request by rjmholt**
Fixes https://github.com/PowerShell/vscode-powershell/issues/1994

### CODE REVIEW

**Comment by rjmholt** [comment]
Wait, it looks like we already have a test for a path like this and it's expected as a possible in-memory input path:

https://github.com/PowerShell/PowerShellEditorServices/blob/fd77c0d85b237f89a532430c130561a04ff5e221/test/PowerShellEditorServices.Test/Session/WorkspaceTests.cs#L163

@rkeithhill is it possible this is already fixed?

**Comment by TylerLeonhardt** [comment]
@rjmholt multiple folks mentioned the debug adapter in the issues... it could be a different code path?

File: src/PowerShellEditorServices/Workspace/Workspace.cs [hunk 1]
```diff
@@ -174,6 +180,20 @@ public ScriptFile GetFile(string filePath)
         /// <param name="scriptFile">The out parameter that will contain the ScriptFile object.</param>
         public bool TryGetFile(string filePath, out ScriptFile scriptFile)
         {
+            try
+            {
+                if (filePath.Contains(":/") // Quick heuristic to determine if we might have a URI
```
Comment (rkeithhill) [inline src/PowerShellEditorServices/Workspace/Workspace.cs]: I thought I had suggested a code change like this before but we eventually decided that the `GetFile()` call on line 199 would throw in this case and we'd return false.  That said, I'd rather test this in a way that doesn't rely on catching an exception.

**Comment by rjmholt** [comment]
> multiple folks mentioned the debug adapter in the issues... it could be a different code path?

The stack trace I'm working against looks like this:

```
    System.AggregateException: One or more errors occurred. ---> System.ArgumentException: Illegal characters in path.
       at System.Security.Permissions.FileIOPermission.EmulateFileIOPermissionChecks(String fullPath)
       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access)
       at Microsoft.PowerShell.EditorServices.Workspace.GetFile(String filePath)
       at Microsoft.PowerShell.EditorServices.Workspace.TryGetFile(String filePath, ScriptFile& scriptFile)
       at Microsoft.PowerShell.EditorServices.Protocol.Server.DebugAdapter.<HandleSetBreakpointsRequest>d__30.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.MessageDispatcher.<DispatchMessage>d__7.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.ProtocolEndpoint.<ListenForMessages>d__36.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.PowerShell.EditorServices.Utility.AsyncContext.Start(Func`1 asyncMainFunc, ILogger logger)
       at System.Threading.Tasks.Task.Execute()
       --- End of inner exception stack trace ---
    ---> (Inner Exception #0) System.ArgumentException: Illegal characters in path.
       at System.Security.Permissions.FileIOPermission.EmulateFileIOPermissionChecks(String fullPath)
       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access)
       at Microsoft.PowerShell.EditorServices.Workspace.GetFile(String filePath)
       at Microsoft.PowerShell.EditorServices.Workspace.TryGetFile(String filePath, ScriptFile& scriptFile)
       at Microsoft.PowerShell.EditorServices.Protocol.Server.DebugAdapter.<HandleSetBreakpointsRequest>d__30.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.MessageDispatcher.<DispatchMessage>d__7.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.ProtocolEndpoint.<ListenForMessages>d__36.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.PowerShell.EditorServices.Utility.AsyncContext.Start(Func`1 asyncMainFunc, ILogger logger)
       at System.Threading.Tasks.Task.Execute()<---
```

So `Workspace.TryGetFile` seems like the right place. What I don't understand is that we seem to have already dealt with this kind of URI scheme elsewhere, but we're still throwing here...

File: src/PowerShellEditorServices/Workspace/Workspace.cs (hunk 1 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Workspace/Workspace.cs]: Sorry @rkeithhill, are you for this change?

Comment (rkeithhill) [reply src/PowerShellEditorServices/Workspace/Workspace.cs]: Yes.

Comment (rjmholt) [reply src/PowerShellEditorServices/Workspace/Workspace.cs]: My worry is that there seems to be a test that exists already for a `git:/` schemed file. My change would just reject those (and anything else that isn't `file`, `untitled` or `inmemory`). The fact that we already have a test for this makes me feel like there is an expectation that we can understand and handle a URI like this -- does that seem right?

Comment (rkeithhill) [reply src/PowerShellEditorServices/Workspace/Workspace.cs]: Those tests for `git:/` and `gitlens-git:/` look like my tests.  I think the intent here was to have those uri schemes return true for IsInMemory which we used to signal that you can't do things like set breakpoints with an "in memory" document.  In retrospect, perhaps that is not the best approach RE the way we used IsInMemory but the basic idea was to determine whether or not we had a file path we could provide to `Set-PSBreakpoint`.

Comment (rkeithhill) [reply src/PowerShellEditorServices/Workspace/Workspace.cs]: In an ideal design, I could walk up to the ScriptFile/EditorDocument and query a property like `CanHaveBreakpoints` or something like that to determine whether or not an LSP text document could have a breakpoint set on it.  But I think this is good as a stop-gap.

**Pull request merged**